terraform/deploy

```diff
@@ -0,0 +1,73 @@
+#!/bin/sh
+
+set -eu
+
+if [ $# -lt 1 ]; then
+    echo "usage: $0 [ENVIRONMENT] [COMMAND ...]"
+    exit 1
+fi
+
+ENV=${1:-staging}
+shift
+
+conf=$(dirname $0)/deploy.conf
+if [ -r "$conf" ]; then
+    . "$conf"
+fi
+if [ -z "${aws_profile:-}" ] || [ -z "${aws_mfa_serial:-}" ]; then
+    echo "E: Both \$aws_profile and \$aws_mfa_serial must be set."
+    echo "I: Either set them in the environment, or in $conf"
+    exit 1
+fi
+
+# data
+role=arn:aws:iam::495570029050:role/qa-admin
+duration=3600
+
+# assume the qa-admin role
+if [ -e .admin_credentials ]; then
+    now=$(date +%s)
+    credentials_created=$(stat --format=%Y .admin_credentials)
+    if [ $((now - credentials_created)) -gt 3600 ]; then
+        rm -f .admin_credentials
+    fi
+fi
+if [ ! -e .admin_credentials ] || [ ! -s .admin_credentials ]; then
+    read -p "MFA token: " token
+    touch .admin_credentials
+    chmod 600 .admin_credentials
```

Comment:
Could we keep this out of the git repo (even if it's gitignored), for safety? Or better, read/write it to a variable so it never touches the disk?

Reply:
we could, but then if you are working on if you have to enter MFA codes all the time.

Reply:
Ah, right, well how about putting it somewhere in ~/?